polaris/dataset/_dataset.py

```diff
@@ -143,7 +143,7 @@ def _validate_model(cls, m: "Dataset"):
         # Verify the checksum
         # NOTE (cwognum): Is it still reasonable to always verify this as the dataset size grows?
         actual = m.md5sum
-        expected = cls._compute_checksum(m.table)
+        expected = cls._compute_checksum(m.table, m.zarr_root_path)
```

Comment:
We now always compute the checksum. I don't think that makes sense. 

We should probably only do this on up- and download? 

For example, a TB-sized datasets could be stored on the Hub. Do you really want to compute the checksum when you're loading from the hub?

Reply:
I meant to say that the mechanism I'm adding with this PR is for an _entire_ Zarr archive. However, if you don't download the dataset in its entirety but rather keep it stored on the Hub and only load the Zarr chunks you need, it is a waste to compute the checksum for the entire archive.

Reply:
> Do we have any estimation of how many chunks a user would typically need to call in at once?

This depends on how the dataset is chunked and the data access pattern. I don't think we need to worry about this too, too much though. This is abstracted away through Zarr and it seems to me that we just need to make sure that as we download a `chunk` from the Hub through the [`PolarisFileSystem`](https://github.com/polaris-hub/polaris/blob/main/polaris/hub/polarisfs.py), that we verify the hash.